internal/handlers/startup_handler.go

```diff
@@ -0,0 +1,96 @@
+package handlers
+
+import (
+	"backend/internal/middleware"
+	"backend/internal/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type StartupHandler struct {
+	db *gorm.DB
+}
+
+func NewStartupHandler(db *gorm.DB) *StartupHandler {
+	return &StartupHandler{db: db}
+}
+
+func (h *StartupHandler) Register(r *gin.RouterGroup) {
+	startup := r.Group("/startup")
+	{
+		// Public endpoints (require auth)
+		startup.GET("", h.List)
+		startup.GET("/:id", h.Get)
+
+		// Admin-only endpoints
+		admin := startup.Group("/admin")
```

Comment:
same as above: no need to group with sepearate base URL for admin routes, just make sure the middleware `AdminRequired()` is there for the POST, PUT and DELETE endpoints